### ISSUE TITLE
Possible inconsistency in Amylase `one_to_many` split between the paper, split-generation code, and released CSV

### ISSUE THREAD

**Issue by anStudent552** [J-SNACKKB/FLIP#28]
Hi FLIP2 authors,

Thank you for releasing FLIP2 and the associated code and datasets.

I am currently trying to reproduce the **Ridge (one-hot)** baseline for the Amylase `one_to_many` split, and I noticed a possible inconsistency between:

1. the `1-to-many` definition in the paper,
2. the split-generation code in the repository, and
3. the currently released `one_to_many.csv.gz`.

I would greatly appreciate some clarification on which split was used for the results reported in the paper.

## 1. `1-to-many` definition in the paper

The paper describes the Amylase dataset as follows:

> "The dataset includes 3,706 variants with up to eight mutations, of which 488 are single mutants, providing an opportunity to evaluate generalization for an industrially important enzyme."

More importantly, the Amylase `1-to-many (number)` split is explicitly described as:

> "1-to-many (number): Train on variants with 1 mutation; test on all other variants (including the wild type)."

Based on this definition, I would expect the **488 single mutants** to be on the training/validation side, while the WT and multi-mutants would be on the test side.

## 2. Mutation distribution in the currently released `one_to_many.csv.gz`

I downloaded the Amylase `one_to_many.csv.gz` from the FLIP2 website and verified its MD5 checksum against the published checksum:

```text
97006840cd899e93fc011202b7d008c2
```

The file contains **3,706 variants**.

Using an inferred WT sequence and counting the number of substitutions relative to it, I obtained:

| Number of mutations | Number of variants |
|---:|---:|
| 0 | 1 |
| 1 | 488 |
| 2 | 1,047 |
| 3 | 1,065 |
| 4 | 674 |
| 5 | 310 |
| 6 | 93 |
| 7 | 26 |
| 8 | 2 |

The total number of single mutants (**488**) agrees with the number reported in the paper.

However, the `set` labels in the released CSV give the following mutation distribution:

### `set=train`

| Number of mutations | Number of variants |
|---:|---:|
| 0 | 1 |
| 2 | 1,047 |
| 3 | 1,065 |
| 4 | 674 |
| 5 | 310 |
| 6 | 93 |
| 7 | 26 |
| 8 | 2 |
| **Total** | **3,218** |

### `set=test`

| Number of mutations | Number of variants |
|---:|---:|
| 1 | 488 |
| **Total** | **488** |

Therefore, in the currently released CSV:

- all **488 single mutants** are assigned to `set=test`;
- the **WT and all 3,217 multi-mutants** are assigned to `set=train`.

This appears to be the reverse of the `1-to-many` definition given in the paper.

After applying the `validation` flag, I obtain:

```text
train:       2574
validation:   644
test:         488
```

I also checked the mutation counts of the **644 validation variants**:

| Number of mutations | Number of validation variants |
|---:|---:|
| 2 | 222 |
| 3 | 212 |
| 4 | 128 |
| 5 | 58 |
| 6 | 21 |
| 7 | 3 |

Thus, the validation set in the currently released file also consists entirely of multi-mutants.

## 3. The split-generation code appears to define the opposite direction

In `collect_splits/amy_bacsu.py`, I found the following code for generating `one_to_many.csv`:

```python
np.random.seed(0)
df['validation'] = False

for i, row in df.iterrows():
    if row['n_mutations'] > 1:
        df.loc[i, 'set'] = "test"
    else:
        df.loc[i, 'set'] = "train"
        if np.random.random() < 0.15:
            df.loc[i, "validation"] = True
```

As I understand it, this assigns:

```text
0/1 mutations  -> train side
>1 mutations   -> test side
```

This is broadly consistent with the intended one-to-many direction described in the paper.

However, there appears to be a small difference regarding the WT: the paper explicitly states that the WT should be included in the test set, whereas the code above appears to place the WT (`n_mutations == 0`) on the training side.

More importantly, this split-generation logic also appears to be inconsistent with the currently released CSV, where the single mutants are in the test set and the multi-mutants are in the training set.

## 4. Ridge one-hot baseline reproduction using the released CSV

To check whether the discrepancy in the reported result might simply be caused by differences in the Ridge implementation, I reproduced the Ridge one-hot baseline following `baselines/linear_models.py`.

I kept the released CSV unchanged and used its existing `set` and `validation` labels.

Following the baseline code, I used:

```text
training samples:
    set == "train" and validation == False

target scaler:
    all samples with set == "train"

test samples:
    set == "test"
```

and the Ridge configuration:

```python
Ridge(
    alpha=10,
    solver="auto",
    tol=1e-5,
    max_iter=1000000
)
```

This gives:

```text
Dataset size: 3706

Train size:       2574
Validation size:   644
Test size:         488

Validation Spearman: 0.672577
Test Spearman:       0.696423
```

Therefore, even after aligning the Ridge implementation with `baselines/linear_models.py`, the test Spearman remains approximately **0.696**.

This appears to differ substantially from the reported **Ridge (one-hot)** Spearman result for the Amylase `one_to_many` benchmark (approximately **0.43**).

## Questions

Could you please clarify the following?

1. **What is the intended direction of the Amylase `one_to_many` split?**
   - single mutants → multi-mutants, as described in the paper; or
   - multi-mutants → single mutants, as represented by the currently released CSV?

2. **Which version of the Amylase `one_to_many` split was used to generate the baseline results reported in the paper?**

3. **Is the currently released `one_to_many.csv.gz` the same split that was used for the reported Ridge and PLM baseline results?**

4. **Where should the WT be assigned in the intended split?**
   The paper places it in the test set, while `collect_splits/amy_bacsu.py` appears to place it on the training side.

5. **Is there a specific dataset/code version or commit that you recommend using to reproduce the Amylase results reported in the paper?**

I would be happy to provide my reproduction script or any additional diagnostics if useful.

Thank you very much for your time and for releasing the benchmark!
